src/components/controls/datepicker/DatePicker.js

```diff
@@ -0,0 +1,324 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import getYear from 'date-fns/get_year';
+import getMonth from 'date-fns/get_month';
+import getDay from 'date-fns/get_day';
+import getDate from 'date-fns/get_date';
+import startOfMonth from 'date-fns/start_of_month';
+import lastDayOfMonth from 'date-fns/last_day_of_month';
+import format from 'date-fns/format';
+import subMonths from 'date-fns/sub_months';
+import isEqual from 'date-fns/is_equal';
+import {
+  chunk,
+  first,
+  last,
+  range,
+  rangeRight,
+  noop
+} from 'lodash';
+
+import { isParseableDate } from '../../util/prop-validators';
+import * as picker from './picker-fns';
+import pickerStateTransitions from './picker-state-transitions';
+import * as dateConstants from './date-constants';
+
+import PickerHeader from './PickerHeader';
+import PickerGrid from './PickerGrid';
+import PickerButton from './PickerButton';
+
+const Picker = styled.div`
+  padding: 5px 10px 10px 10px;
+  width: 300px;
+`;
+
+export default class DatePicker extends React.Component {
+  static propTypes = {
+    /** Function to run once a full date has been selected */
+    dateSelected: PropTypes.func.isRequired,
+    /** Year to start datepicker */
+    startingYear: PropTypes.number,
+    /** Number of years or months to go back */
+    descentAmount: PropTypes.number,
```

Comment:
Is there a different name we could use for this? Though now that I look at it, it might be hard to come up with a good name since it could either be years or months. Maybe lookback amount?